### DISCUSSION TITLE
ioBroker

### DISCUSSION THREAD

**Discussion by Wlad2288** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#111] [General]
Hi Piotr,

First of all thanks for your amazing work!
I have a Xiaomi RoboRock Gen 1 with the stock ROM integrated in my ioBroker instance and everything works fine except I can't get the map without rooting and flashing it with another firmware.

Then I found your Cloud Map Extractor and it seems to be exactly what I am looking for.
However, I am not using Home Assistant :(

Do you think this could be integrated with ioBroker as well?

Many Thanks,
Wlad

**Comment by PiotrMachowski** [comment]
Hi Wlad,

I don't use ioBroker, so my answer might not be correct, but I have made a quick research and ioBroker is written on Node.js, so this tool would probably have to be rewritten in a different language (e.g. JavaScript, TypeScript). My implementation was based on two existing ones: [OpenHAB](https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud) (written in Java) and  [valeCLOUDo](https://github.com/Xento/valeCLOUDo/blob/testing/lib/SmarthomeAPI.js) (written in JS) - this one might be a solution for your problem.

You are also welcome to migrate to Home Assistant 😃 

Best regards,
Piotr

**Reply by Wlad2288** [reply]
Thanks for your help but I fear I won't be able to do it with my limited scripting skills and time :)
I just saw you also developed a tool that grabs the map from the vacuum via ssh.
I think that might be an option for me, as enabling SSH won't break the cloud connection right?

Do you know the location in the vacuum Ubuntu distro, where the map files are located?

**Reply by PiotrMachowski** [reply]
[here](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Vacuum-Map-Camera/blob/master/custom_components/xiaomi_vacuum_map/camera.py#L268) you have a script that retrieves data required for map generation. Unfortunately these files have to be parsed to convert them to an image, so it is not a perfect solution.
If you are willing to root your vacuum then [this](https://macbury.github.io/SmartHouse/HomeAssistant/Vacuum/) (code: [link](https://github.com/macbury/SmartHouse/tree/master/vacuum)) might be a better solution for you :)
